src/KafkaNetClient/Consumer.cs

```diff
@@ -226,6 +226,17 @@ private Task ConsumeTopicPartitionAsync(string topic, int partitionId)
                             needToRefreshMetadata = true;
                             _options.Log.ErrorFormat(ex.Message);
                         }
+                        catch (NoLeaderElectedForPartition ex)
+                        {
+                            needToRefreshMetadata = true;
+                            _options.Log.ErrorFormat(ex.Message);
+                        }
+                        catch (LeaderNotFoundException ex)//the numbar partition of can be change
+                        {
+                            needToRefreshMetadata = true;
+                            _options.Log.ErrorFormat(ex.Message);
+                        }
+
```

Comment:
Suggestion: http://geekswithblogs.net/BlackRabbitCoder/archive/2015/04/09/c.net-little-wonders-exception-filtering-in-c-6.aspx

Just an idea